internal/backend/runtime/omni/controllers/omni/cluster_machine_config_status.go

```diff
@@ -587,19 +604,23 @@ func (h *clusterMachineConfigStatusControllerHandler) reset(
 	}
 
 	// if is control plane first leave etcd
-	if graceful && isControlPlane {
-		_, err = c.EtcdForfeitLeadership(ctx, &machineapi.EtcdForfeitLeadershipRequest{})
-		if err != nil {
-			h.ongoingResets.forceUngraceful(clusterMachine.Metadata().ID())
+	if isControlPlane && !h.ongoingResets.skipLeave(clusterMachine.Metadata().ID()) {
```

Comment:
this is the part of the fix to make ungraceful reset less aggressive.

`skipLeave` and `isGraceful` are now two separate flags instead of a single one.
If etcd leave attempt fails Omni will still try to do graceful reset attempt, but the next reset call will skip the etcd leave call.